### PR TITLE
Add end time to appointment notifications (and refactor a bit)

### PR DESCRIPTION
**Pull request by Jonasdoubleyou**

### CODE REVIEW

File: common/appointment/util.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    return {
+        ...appointment,
+
+        start_day: appointment.start.toLocaleString(language, { weekday: 'long' }),
+        start_date: appointment.start.toLocaleString(language, { day: 'numeric', month: 'long', year: 'numeric' }),
+        start_time: appointment.start.toLocaleString(language, { hour: '2-digit', minute: '2-digit' }),
+
+        end_date: end.toLocaleString(language, { day: 'numeric', month: 'long', year: 'numeric' }),
+        end_time: end.toLocaleString(language, { hour: '2-digit', minute: '2-digit' }),
```
Comment (Jonasdoubleyou) [inline common/appointment/util.ts:23]: The actual change, add end_date and end_time here

File: common/appointment/cancel.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    await prisma.lecture.update({
+        data: { isCanceled: true },
+        where: { id: appointment.id },
+    });
+
+    logger.info(`Appointment(${appointment.id}) was cancelled by User(${user.userID})`);
+
+    const student = await getStudent(user);
+
+    switch (appointment.appointmentType) {
```
Comment (Jonasdoubleyou) [inline common/appointment/cancel.ts:20]: replaced if...else by switch to later turn on exhaustiveness checking for switch on union types

File: common/appointment/cancel.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const logger = getLogger('Appointment');
+
+export async function cancelAppointment(user: User, appointment: Appointment) {
+    await prisma.lecture.update({
+        data: { isCanceled: true },
+        where: { id: appointment.id },
+    });
+
+    logger.info(`Appointment(${appointment.id}) was cancelled by User(${user.userID})`);
```
Comment (Jonasdoubleyou) [inline common/appointment/cancel.ts:16]: added logging

File: common/appointment/update.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    const updatedAppointment = await prisma.lecture.update({
+        where: { id: appointment.id },
+        data: { ...appointmentUpdate },
+    });
+
+    logger.info(`User(${user.userID}) updated Appointment(${appointment.id})`, { appointment, appointmentUpdate });
+
+    const sameDate = !appointmentUpdate.start || appointment.start.toISOString() === appointmentUpdate.start.toISOString();
```
Comment (Jonasdoubleyou) [inline common/appointment/update.ts:26]: Added check whether start is actually updated

Comment (timidak) [reply common/appointment/update.ts:30]: Nice work 🎉

**Review by timidak — APPROVED** [review]
This cleaned up the code quite a bit!
Nice work 🎉

**Pull request merged**